### PR TITLE
quarkus grpc - multiple server instances

### LINKED ISSUE #10057: No options to scale gRPC server
**Describe the bug**
`io.vertx.grpc.VertxServer` do not auto-scale across Vert.x event-loop. In Vert.x the scaling is determined by the number of deployed instances of a Verticles. I am not entirely familiar with how Quarkus initializes the `GrpcServerRecorder`, but it seems like only one instance of `VertxServer` is ever started.

**Expected behavior**
Singletons implementing a gRPC Service should be allowed to scale across Vert.x event-loop. Additionally, provide a way to configure the desired scaling.

**Actual behavior**
At most 2 Vert.x event loop threads process request for `VertxServer`

**To Reproduce**
1. Implement a Quarkus gRPC server with one service annotated with `@Singleton` and extending `MutinyApiConfigServiceGrpc.[GrpcServiceName]ImplBase`
2. In the service, write a log line to print the Thread Name
3. Build and run the server
4. Make RPC calls. Used ghz.sh to simulate load.
5. Notice in the logs that only 2 unique thread names are printed. Threads will have the prefix `vert.x-eventloop-thread-`.

**Configuration**
No additional configuration were provided. gRPC extension was using the default values.

**Environment (please complete the following information):**
- CPU count: 8
- Vert.x Event Loop thread count: 16 (the default 2*CPU)
 - Output of `uname -a` or `ver`:
```
Darwin REMMACJ5Y8GTFM 19.5.0 Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
```
 - Output of `java -version`:
```
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```
 - Quarkus version or git rev: 1.5.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
------------------------------------------------------------
Gradle 6.5
------------------------------------------------------------

Build time:   2020-06-02 20:46:21 UTC
Revision:     a27f41e4ae5e8a41ab9b19f8dd6d86d7b384dad4

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.2 (Oracle Corporation 11.0.2+9)
OS:           Mac OS X 10.15.5 x86_64
```

**Additional context**
`io.quarkus.grpc.runtime.GrpcServerRecorder` has the following comment on line 62

```
// TODO Support scalability model (using a verticle and instance number)
```

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #10057

### CODE REVIEW

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcServerConfiguration.java [hunk 1]
```diff
@@ -65,4 +65,10 @@
      */
     @ConfigItem(defaultValue = "false")
     public boolean enableReflectionService;
+
+    /**
+     * Number of gRPC server verticle instances.
```
Comment (cescoffier) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcServerConfiguration.java:70]: I would add a sentence about the scaling. It's not obvious.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Documentation should also be updated to mention the scaling mechanism.

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcServerConfiguration.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcServerConfiguration.java:70]: Updated

**Review by cescoffier — APPROVED** [review]

File: docs/src/main/asciidoc/grpc-service-implementation.adoc [hunk 2]
```diff
@@ -179,6 +179,11 @@ This service allows tools like https://github.com/fullstorydev/grpcurl[grpcurl]
 The reflection service is enabled by default in _dev_ mode.
 In test or production mode, you need to enable it explicitly by setting `quarkus.grpc-server.enable-reflection-service` to `true`.
 
+== Scaling
+By default, a quarkus-grpc starts a single gRPC server running on a single event loop.
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-service-implementation.adoc:183]:
```suggestion
By default, quarkus-grpc starts a single gRPC server running on a single event loop.
```

**Review by gsmet — DISMISSED** [review]
Spotted a small typo. Feel free to dismiss as soon as it's fixed!

**Pull request merged**